projects/core/src/product/store/selectors/product-references.selectors.ts

```diff
@@ -16,13 +16,18 @@ export const getProductReferencesState: MemoizedSelector<
 );
 
 export const getSelectedProductReferencesFactory = (
-  productCode: string
+  productCode: string,
+  referenceType: string
 ): MemoizedSelector<StateWithProduct, ProductReference[]> => {
   return createSelector(
     getProductReferencesState,
     referenceTypeData => {
       if (referenceTypeData.productCode === productCode) {
-        return !!referenceTypeData.list ? referenceTypeData.list : [];
+        return !!referenceTypeData.list
+          ? referenceTypeData.list.filter(
+              item => item.referenceType === referenceType
```

Comment:
In the `service`, `referenceType` is optional. So, I think, if `referenceType` is undefined, maybe we can return all references?